gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -234,6 +238,26 @@ private boolean isDirectPathEnabled(String serviceAddress) {
     return false;
   }
 
+  // DirectPath should only be used on Compute Engine.
+  // Notice Windows is supported for now.
+  public static boolean isOnComputeEngine() {
```

Comment:
Does this need to be public? If it's just for testing we can make it default visibility.

Reply:
This is a good point. public has been removed. Thanks!